apps/documents/views.py

```diff
@@ -206,31 +205,45 @@ def form_valid(self, form):
 class DeleteCollection(LoginAndTeamRequiredMixin, View):
     def delete(self, request, team_slug: str, pk: int):
         collection = get_object_or_404(Collection, team__slug=team_slug, id=pk)
-        if nodes := collection.get_node_references():
+
+        if collection.archive():
+            messages.success(request, "Collection deleted")
+            return HttpResponse()
+        else:
+            # Find and show references.
+            # For working versions, the Pipelines.
+            # For versions, the experiments
+
+            pipeline_node_chips = [
+                Chip(
+                    label=node.pipeline.name,
+                    url=node.pipeline.get_absolute_url(),
+                )
+                for node in collection.get_related_nodes_queryset()
+            ]
+            experiment_chips = []
+            for version in collection.versions.all():
+                if experiments := version.get_related_experiments_queryset():
+                    experiment_chips.extend(
+                        [
+                            Chip(
+                                label=(
+                                    f"{experiment.name} [{experiment.get_version_name()}]"
+                                    if experiment.is_working_version
+                                    else f"{experiment.name} {experiment.get_version_name()} [published]"
```

Comment:
the version is not necessarily published, right? we just check `self.versions.exists()` not just for published versions

Reply:
Correct. Unless I'm forgetting a reason why we should not, I think this should be the case for assistants as well. Reason being that if a non-published version of some chatbot is using a collection, but we ignore that while archiving said collection, we'll be breaking that version without telling the user or archiving that version. I think we should inform users if a resource like a collection or assistant is being used in any version of a chatbot.

Reply:
I had thought that we'd also archive the version but perhaps we didn't implement that.

As it stands however, this is still incorrect since it's not necessarily a published version because the queryset isn't filtering out unpublished versions.

Reply:
Correct. So the desired behaviour is to archive non-published versions if they are using this collection? (Same goes for assistants, but we don't yet do that)